### PR TITLE
MINIFI-320: Move instantiation of OpenSSL to a singleton that's

### PR DESCRIPTION
**Pull request by phrocker**
lazily created. Added test. In order to make the test work as
expected I had to update thread pool to use a non blocking queue.

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/include/utils/ThreadPool.h [hunk 1]
```diff
@@ -189,7 +195,8 @@ class ThreadPool {
 // atomic running boolean
   std::atomic<bool> running_;
 // worker queue of worker objects
-  std::queue<Worker<T>> worker_queue_;
+  //std::queue<Worker<T>> worker_queue_;
```
Comment (phrocker) [inline libminifi/include/utils/ThreadPool.h]: remove comment

File: libminifi/test/unit/SocketTests.cpp [hunk 2]
```diff
@@ -183,3 +196,42 @@ TEST_CASE("TestSocketWriteTestAfterClose", "[TestSocket6]") {
...
+  std::this_thread::sleep_for(std::chrono::milliseconds { distribution(seed) });
+
+  for (int i = 0; i < 50; i++) {
+    std::shared_ptr<TLSContext> socketA = std::make_shared<TLSContext>(
+        configuration);
+    socketA->initialize();
+  }
+
+  return true;
+}
```
Comment (phrocker) [inline libminifi/test/unit/SocketTests.cpp]: add some comments to why this exists.

**Comment by phrocker** [comment]
@apiri this PR is ready for review.

**Comment by apiri** [comment]
reviewing

**Comment by apiri** [comment]
Code, build, and tests all look good.  Will get this merged.

**Pull request closed without merging**